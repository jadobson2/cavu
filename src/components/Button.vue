<template>
  <button
    class="
      px-8 py-3 rounded-xl bg-yellow hover:bg-yellow-light text-lg font-bold
      transition-colors disabled:bg-greyscale-20 disabled:text-greyscale-80
      disabled:opacity-50
    "
    @click="$emit('click', $event)"
  >
    <slot></slot>
  </button>
</template>

<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
  // eslint-disable-next-line vue/multi-word-component-names
  name: 'Button'
})
</script>
