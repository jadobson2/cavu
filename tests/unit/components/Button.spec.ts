import { shallowMount } from '@vue/test-utils'
import Button from '@/components/Button.vue'

describe('Button', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(Button, {
      slots: { default: 'Foo' }
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  test('click', () => {
    const wrapper = shallowMount(Button, {
      slots: { default: 'Foo' }
    })
    wrapper.trigger('click')

    expect(wrapper.emitted().click).toBeTruthy()
    expect(wrapper.emitted().click?.[0][0]).toEqual(expect.any(Event))
  })
})
