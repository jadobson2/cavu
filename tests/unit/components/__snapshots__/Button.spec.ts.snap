// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  class="px-8 py-3 rounded-xl bg-yellow hover:bg-yellow-light text-lg font-bold transition-colors disabled:bg-greyscale-20 disabled:text-greyscale-80 disabled:opacity-50"
>
  Foo
</button>
`;
